lib/ok.ex

```diff
@@ -83,6 +83,38 @@ defmodule OK do
     if is_list(result), do: {:ok, Enum.reverse(result)}, else: result
   end
 
+  @doc """
+  Takes a result tuple, a predicate function, and an optional error reason.
+  If the result tuple is tagged as a success then its value will be passed to the predicate function.
+  If the predicate returns `true`, then the result tuple stay the same.
+  If the predicate returns `false, then the result tuple becomes `{:error, reason}`.
+  If the tag is failure then the predicate function is skipped.
+
+  ## Examples
+
+      iex> OK.filter({:ok, 2}, fn (x) -> x == 2 end)
```

Comment:
Good examples. :-)

I think that these examples are run as tests so we don't need to duplicate the tests in the test file.
Also could you add one final example where the filter fails but uses the standard return reason

Reply:
I'll do that. By the way, the Elixir unit test system and its ex doc integration is pretty awesome.